### DISCUSSION TITLE
Feature Request: loudness normalization for song files according to EBU R 128 guidelines

### DISCUSSION THREAD

**Discussion by ampersand64** [ppy/osu#23517] [Ideas] [answered]
This is a quality of life feature. Normalizing song volume to -24 LUFS or whereabouts has many benefits. 
Normalizing to this lower volume would attenuate more songs, meaning a more consistent loudness throughout the game. 
I guarantee, almost no one is having problems with osu! being too quiet. My master volume usually sits around 50%. 

As it stands, song volume inconsistencies force users to adjust the relative volumes of effects and music if they want to balance hitsound volume. This degrades user experience.

This feature would be particularly useful on maps with the same song with different volumes, such as these: 
https://osu.ppy.sh/beatmapsets/1500436#osu/3074715
https://osu.ppy.sh/beatmapsets/1477257#osu/3030980
This is likely to cause problems, since sorting by artist or alphabetically will place these two maps next to each other. 

It also mitigates any potential problem for audio clipping due to poor encoding. 
Using loudness units will keep the _perceived_ volumes of songs with different spectral content and dynamic range consistent. These two songs, for example, are at opposite ends of both spectrums:
https://osu.ppy.sh/beatmapsets/410162#osu/890190
https://osu.ppy.sh/beatmapsets/1337087#osu/2769855

It could be possible to write a replay gain tag for each beatmap's audio (or something similar in the beatmap file), then apply the gain upon import of the beatmap. 

I am not qualified to talk about code or game design. But you get the picture.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/issues/1600

**Reply by ampersand64** [reply]
Thank you. Sorry I missed this.
